### PR TITLE
improve developer ergonomics, based on feedback from internal demo

### PR DESCRIPTION
**Pull request by skiggety**

### CODE REVIEW

**Review by bistline — APPROVED** [review]
I like the readability changes - that should make things a little more explicit.

File: lib/bash_include/dockerhub.bash [hunk 1]
```diff
@@ -9,7 +9,7 @@ fi
 # making sure publish fails if version.txt has not been updated to avoid re-using the version tag:
 function assert_main_version_unused {
     if main_version_already_published; then
-        exit_with_error_message "Version $MAIN_VERSION has already been published, and refers to a different image. Please edit version.txt according to semver and try again."
+        exit_with_error_message "Version $MAIN_VERSION has already been published, and refers to a different image. Something must have changed, perhaps a newer version of $BASE_IMAGE (BASE_IMAGE). Please edit version.txt according to semver and try again to publish these changes."
```
Comment (eweitz) [inline lib/bash_include/dockerhub.bash]: Even clearer:

```suggestion
        exit_with_error_message "Version $MAIN_VERSION has already been published, and refers to a different image. Something must have changed, perhaps a newer version of $BASE_IMAGE (BASE_IMAGE). Please edit version.txt in https://github.com/broadinstitute/scp-rails-baseimage according to semver and try again to publish these changes."
```

**Review by eweitz — APPROVED** [review]
Thanks for the comprehensive walk-through, and incorporating feedback.

~In the failure email, or perhaps one of the error messages you showed today, could you add a link to the [SCP Playbook entry on "Updating Rails managed base image"](https://docs.google.com/document/d/1vJz6hXdjAu8LBAfIS4oMy7yLIUX177gU1H7BxTPPzAU/edit#heading=h.8gg7zdig7ttm)?  Those summary directions would help the rotating release engineer.~  <-- Done while I was reviewing (quick! :))

**Review by jlchang — APPROVED** [review]
Thanks for today's walkthrough. I appreciate having the additional info to guide future maintenance tasks.

File: lib/bash_include/dockerhub.bash (hunk 1 above)
Comment (skiggety) [reply lib/bash_include/dockerhub.bash]: Just committed a non-hardcoded version of this suggestion.

**Comment by skiggety** [comment]
Thanks, all, for the participation and feedback! (now let's hope I can get the build to pass again)

**Pull request merged**